File: VB/CustomTokenEdit/CustomTokenEdithandler.vb
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace Example.CustomTokenEdit
	Public Class CustomTokenEdithandler
		Inherits TokenEditHandler

		Public Sub New(ByVal tokenEdit As TokenEdit)
			MyBase.New(tokenEdit)
		End Sub

		Protected Overrides Sub OnTokenClick(ByVal hitInfo As TokenEditHitInfo)
			ChangeCheckedIfRequired(hitInfo)
			MyBase.OnTokenClick(hitInfo)
		End Sub

		Private Sub ChangeCheckedIfRequired(ByVal hitInfo As TokenEditHitInfo)
			Dim hInfo As CustomTokenEditHitInfo = TryCast(hitInfo, CustomTokenEditHitInfo)
			If hInfo.IsInCheckBox Then
				TryCast(hitInfo.HitObject, CustomTokenEditTokenInfo).ChangeChecked()
			End If
		End Sub
	End Class
End Namespace

File: CS/TokenEdit_Glyph_InEndOfToken/CustomTokenEdit/CustomTokenEdithandler.cs
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ViewInfo;
using System;
using System.Collections.Generic;
using System.Linq;

namespace TokenEdit_Glyph_InEndOfToken.CustomTokenEdit
{
    public class CustomTokenEdithandler : TokenEditHandler
    {
        public CustomTokenEdithandler(TokenEdit tokenEdit) : base(tokenEdit) { }

        protected override void OnTokenClick(TokenEditHitInfo hitInfo)
        {
            ChangeCheckedIfRequired(hitInfo);
            base.OnTokenClick(hitInfo);
        }

        private void ChangeCheckedIfRequired(TokenEditHitInfo hitInfo)
        {
            CustomTokenEditHitInfo hInfo = hitInfo as CustomTokenEditHitInfo;
            if (hInfo.IsInCheckBox)
                (hitInfo.HitObject as CustomTokenEditTokenInfo).ChangeChecked();
        }
    }
}
